### DISCUSSION TITLE
Duplicate chars in vs2022 code window

### DISCUSSION THREAD

**Discussion by jlgarcia-vs** [community/community#14206] [Copilot Conversations]
Hi,

I've a strange problem in one of the code windows of vs2022.
With the GitHub Copilot activated, every time I press a key, six chars are typed, like this example "ddddddeeeeeebbbbbbuuuuuugggggg".

This problem just happens in one of my code files (asp.net mvc, .cshtml, net framework 4.6.1). Other cshtml files work fine.

Regards
Jose

**Comment by Edddy** [comment]
Same here, at least I'm not crazy

**Reply by Edddy** [reply]
Follow up. I've enabled HTML projections to see if the problem was solved.

The problem appears to triggers only in some files, not all .cshtml

https://user-images.githubusercontent.com/854168/179003299-7e8481c2-824b-437b-a2a1-66cc77a52802.mp4

**Comment by hallatore** [comment]
I have 4 in mine. Happens in cshtml files

**Reply by sotework** [reply]
me too in vue files

**Comment by Pretagonist** [comment]
Have same issue. In cshtml file in VS 2022 I get double letters, inside an @ code block I get quadruple. 
Could it be a conflict with ReSharper?

**Reply by Edddy** [reply]
I don't use ReSharper, so I don't think thats the problem.

**Comment by jlgarcia-vs** [comment]
@Pretagonist I don't think it's a problem with ReSharper. I'm not using ReSharper in my VS2022.
Regarding the number of letters, yes, I'm getting quadruple also.
Regards

**Comment by hallatore** [comment]
![image](https://user-images.githubusercontent.com/365605/162144285-3b0b6dcc-0aa7-4b6a-ad81-ef36ebf8dd96.png)

A workaround for now is turning off "**Enable for HTMLProjection**" and reopening your .cshtml file.

**Reply by Diminios** [reply]
Thank you. This solved the problem for me - at least it looks like it, at the moment.

**Reply by yanxiaodi** [reply]
Thanks. It's solved.

**Reply by grovolis** [reply]
Thanks, it works for me.

**Reply by OdedDvir** [reply]
Thank you. it worked!

**Reply by Xaninho** [reply]
Worked here as well!

**Reply by tamar-martin** [reply]
How do you get to this window?

**Reply by thomasalfaedison** [reply]
How to get that options?

**Comment by mfsbo** [comment]
The problem was only in typing code and I uninstalled liveshare and github copilot from extensions on a fresh install of Visual Studio 2022 17.3.1.
the issue has been raised and closed on VS code multiple times so just linking it here
https://developercommunity.visualstudio.com/t/Pressing-a-key-will-commit-the-key-twice/1464634 and 
https://developercommunity.visualstudio.com/t/visual-studio-16103-editor-double-typing/1464938

Github Copilot Info is:
![image](https://user-images.githubusercontent.com/50488401/162200186-af75b4e0-baf7-4562-8dcc-f26873da8f2f.png)



For Visual Studio Info please see below
```
Microsoft Visual Studio Professional 2022
Version 17.1.3
VisualStudio.17.Release/17.1.3+32328.378
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Professional

Visual C++ 2022   00476-80000-00000-AA416
Microsoft Visual C++ 2022

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   17.1.358.51495
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.1.358.51495
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.1.358.51495
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.1.358.51495
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

GitHub Copilot   1.10.0.3 (v1.10.0.3@764d4f318)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.10.5413 (v1.10.0)

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   17.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.161+abc97ecc7d.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Node.js Tools   1.5.40105.1 Commit Hash:1822ee94b29c6cf748a19825f14cc26d30b0b871
Adds support for developing and debugging Node.js apps in Visual Studio

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2206201+62a2c1d6162f828801565a7ca26d9d48b810a05b
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62203.25080
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.1229.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Inline Suggestions   2.2.1930.216
Ghost text API for Visual Studio inline suggestions

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers
```
System Basic Info is
```
OS Name	Microsoft Windows 10 Pro
Version	10.0.19043 Build 19043
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
```

**Reply by kzu** [reply]
The linked feedback items are from way before the Copilot VS extension even existed, so I doubt they are related in any way 🤔

**Reply by mfsbo** [reply]
Yes thats what I mean, there seem to be issue with visual studio and they did not have enough info to replicate it. I raised today because i only installed vs today latest version and opened and it was ok till I uninstalled visual studio live share and that caused this issue. The only two things I had were live share and github copilot so I disabled copilot and uninstalled live share then no issue for 2 hours coding in html c# and js.
I then enabled gothub copilot at global file and html projections(middle option in 3 enable options) and the problem came back. Then disabled middle option like suggested above and problem went away after restarting vs
My opinion is that issue resides with VS and some bug/poor implementation in VS. When github copilot tries to use that code as expected it does not and instead give double,tripple or some times 4 time output. VS 2022 team should look into it so I raised it there as well https://developercommunity.visualstudio.com/t/Pressing-a-key-will-commit-the-key-twice/10008414 please do provide as much info as possible so they can replicate and resolve it if its not github copilot.

**Comment by samrobley** [comment]
Can re-create the problem by typing an underscore which enables it.
Restarting visual studio clears the toggle.

Only impacted html pages in my experience.
Disabled co-pilot and problem has gone.

**Reply by kzu** [reply]
Could you provide a bit more data for a repro? Ideally, a github repo I can open directly. Please do let me know if you are opening a solution or using the Open Folder functionality from VS. The structure and/or files in the repro might be relevant, which is why a clean repro could be very helpful. Thanks!

**Comment by mfsbo** [comment]
Its opening a solution and you can use starter mvc template and open a chtml view file to edit it.

**Reply by kzu** [reply]
This might be disappointing, but the reason I'm asking it's because I can't reproduce it with a clean install with just VS + ASP.NET + GH Copilot:

![mvc](https://user-images.githubusercontent.com/169707/165654368-10c7c329-7311-48f5-9541-7e77e6c228c8.gif)

**Reply by Edddy** [reply]
In my case I'm on .net Framework 4.8 (not Core). Maybe that's the problem?

**Reply by mfsbo** [reply]
once i deselect "Enable for HTML Projections" it woks fine. I also uninstalled Live Share plugin that comes with Visual Studio 2022. Mine is VS Professional 2022 and I have not tried on any other version. I felt that Live Share and anything that look at key strokes would be an issue and VS team has closed my issue https://developercommunity.visualstudio.com/t/Pressing-a-key-will-commit-the-key-twice/10008414 saying its github copilot. If I enable HTML Projections the issue comes back.

**Reply by mfsbo** [reply]
can you try to type in the code produced and also try to type after suggestions are made and change the code. My problem was not within suggested code but in your case its very less amount of code so I am just trying to figure out how you can replicate. I will try to create a public repo and put some codes there if possible but I am away for 2 weeks.

**Reply by Edddy** [reply]
This is true for me too -> Once I deselect "Enable for HTML Projections" it woks fine.

**Comment by sitefinitysteve** [comment]
I subscribed and just hit this issue myself, just not going to be addressed after it's out of beta and a product we're subscribing to?

**Reply by Edddy** [reply]
Yes right? I just hit the issue but now I'm a paying subscriber

**Reply by jeffschulz** [reply]
I am getting the issue in Microsoft Visual Studio Enterprise 2022 (64-bit) - Current - Version 17.2.3 - VisualStudio.17.Release/17.2.3+32526.322
GitHub Copilot   1.31.0.0 (v1.31.0.0@9b47f3591)
GitHub Copilot Agent   1.31.6194 (v1.31.0)

I am editing javascript in vue files

It worked fine when I first starting using GitHub Copilot and started after about a day or two afterwards.
I disabled GitHub Copilot and restarted VS, it stopped the duplicates from happening
I enabled it again and restarted VS and it was worked fine for two days and today it has started happening again
I have disabled GitHub Copilot and restarted VS and enabled it again and restarted VS and it is working fine now, will try and get a more accurate idea of when it starts duplicating again

**Comment by Kazbek** [comment]


**Reply by RichieCr7** [reply]
Duplicate characters still happening in cshtml pages even after deleting .vs folder

**Comment by RichieCr7** [comment]
This works for anyone still facing this issue in 2023

For my specific issue, I’ve found that disabling “Auto list members” under [Options>Text Editor>HTML>General] seems to resolve the issue.

**Reply by Kazbek** [reply]
This helped on Visual Studio Community 2022 x64 v17.5.1
But for me it was in css files so in Text Editor I selected CSS intead of HTML.

**Comment by stevebolster** [comment]
I am seeing it only within script tags in aspx pages. Anywhere outside of the script tag is fine. VS 2022 17.6.2 with or without CoPilot enabled. Very frustrating..

**Reply by maurox80** [reply]
For me in .cshtml page whitin script tags but only if i use Promise

**Reply by robertomilian-profitsolv** [reply]
I disabled "Web Live Preview" from extensions and it resolved the issue for me.

**Comment by MrChriZ** [comment]
I've just encountered it here.  I found this stackoverflow post:
https://stackoverflow.com/questions/73826648/duplicate-characters-when-typing-in-vs-2022-cs-page#:~:text=This%20issue%20has%20cropped%20up,%2D%3Eget%20tools%20and%20features.&text=so%2C%20usually%20a%20repair%20should%20fix%20this%20issue.

Which pointed out that it only happens after line 89 - and low and behold on my system it's exactly the same.  Fine until line 89...

- I don't have copilot installed on this machine.
- The auto list members option did not work for me.

Weird!

**Reply by paulsplace** [reply]
Hi @MrChriZ I had the same issue earlier today. I discovered that I was trying to add a jquery event underneath a method that contained an @ symbol which is used to switch between markup and code. In my case it was @Url.Action. My workaround was to move the method containing the @Url.Action to the bottom of the script block and I was then able to type freely above it without any duplicate characters. I've added a reply to the stackoverflow post you mentioned and here is the permalink to my findings:  (https://stackoverflow.com/questions/73826648/duplicate-characters-when-typing-in-vs-2022-cs-page/76440470#76440470).

Hope this helps someone (my first post so be kind)!

**Reply by spartan73rmy** [reply]
Hi I had the same issue and temporarily fixed it disabling the Use Legacy Razor Editor for ASP .NET Core 
I have the VS 17.6.3 without copilot 

I was trying with the solutions in the stack overflow but it doesn't work for me
![image](https://github.com/community/community/assets/20409153/62d19500-b8b9-4023-b2bd-ebd9a920a92f)

I don't know why it works, but I hope it helps

**Reply by hieutk10** [reply]
I had updated my visual studio to version 17.6.5
The issue is resolved!

**Comment by robbiexiong** [comment]
as @paulsplace issue, same here. Only in my case, I need to use razor syntax in cshtml, so this is frustrating as I cannot just move a piece of code below everything else when it's needed right where it is being used.

**Comment by MrChriZ** [comment]
See this - looks like MS are aware of it and there is a fix incoming:
https://developercommunity.visualstudio.com/t/Duplicate-key-strokes-when-editing-javas/10372422

**Comment by Edddy** [comment]
It started to happen again in 17.12
17.12.1 does not fixed it

**Reply by Rapidacting** [reply]
You seen any solution?

**Reply by safteinzz** [reply]
Same here, same as old issue. Happening to me in cshtml files and after typing @

Hopefully they release something quick, gotta disable until then seems like.

**Reply by mfsbo** [reply]
Yes I am having same in 17.12 again.

**Reply by mfsbo** [reply]
I was able to accept few suggestions but then later on Github Copilot output gave this when i rejected one in html
```
[CompletionsProposalSource Information] Request completions: |+28/1190: ->"<td>" [1]
[CopilotClientManager Error] [relatedFiles] copilotRelated: ResponseError: Value cannot be null.
Parameter name: uriString
    at handleResponse (C:\snapshot\copilot-client\node_modules\vscode-jsonrpc\lib\common\connection.js:565:48)
    at handleMessage (C:\snapshot\copilot-client\node_modules\vscode-jsonrpc\lib\common\connection.js:345:13)
    at processMessageQueue (C:\snapshot\copilot-client\node_modules\vscode-jsonrpc\lib\common\connection.js:362:17)
    at Immediate.<anonymous> (C:\snapshot\copilot-client\node_modules\vscode-jsonrpc\lib\common\connection.js:334:13)
    at processImmediate (node:internal/timers:471:21) {
  code: -32000,
  data: {
    type: 'System.ArgumentNullException',
    message: 'Value cannot be null.\r\nParameter name: uriString',
    stack: '   at System.Uri..ctor(String uriString)\r\n' +
      '   at Microsoft.VisualStudio.Completions.CopilotLanguageClient.<<AttachForCustomMessageAsync>b__37_7>d.MoveNext()',
    code: -2147467261
  }
}
```

After this it started to write double characters

**Reply by mfsbo** [reply]
I have uninstalled Visual Studio and done a fresh install. Opened an aspx webforms file and start typing some html tags. Typing `<` was ok then adding `tr` to it produced 2 x r and then contiunye however if I do `>` first then move the customer before it and add `<tr` it works and gives hint about `td` next.

```
Conversations Information] Copilot chat version 17.12.31-rc+9db98df8dd.RR (17.12.31.40377). VS: VisualStudio.17.Release/17.12.1+35514.174. Session: 49e172d1-56e9-4bc8-ae55-ba077ec33fed. 
[Conversations Error] User is not signed into GitHub. Please sign into GitHub. https://learn.microsoft.com/en-us/visualstudio/ide/work-with-github-accounts?view=vs-2022
[Conversations Information] Copilot badge status set to: NotSignedInToGitHub
[Conversations Information] Need to get an auth token.
[Conversations Information] [Conversations] Out-of-process service (PID: 34776) finished initializing.
[Conversations Information] Status Code: 200, Reason Phrase: OK
[Conversations Information] Response Content: HasToken: True, ChatEnabled: True, ExpiresAt: 1732729851, RefreshIn: 1500, SKU: yearly_subscriber, CopilotExclusion: False
[Conversations Information] Copilot badge status set to: OK
[CopilotClientManager Information] Auth status changed: OK
[Conversations Information] Copilot badge status set to: OK
[Conversations Information] [ChunkIndex] Index db path: C:\Root\Projects\repo\.vs\somefolder-System\CopilotIndices\17.12.31.40377\CodeChunks.db
[CopilotClientManager Information] LSP: Activated (65162350)
[CopilotClientManager Information] LSP: Debug Server: http://localhost:8180
[CopilotClientManager Information] LSP: Process 38252
[CopilotClientManager Information] Initial auth status: OK
[CompletionsProposalSource Information] Request completions: |+24/2: No completions found
[CompletionsProposalSource Information] Request completions: 						<|+0/3: No completions found
[CompletionsProposalSource Information] Request completions: 						<|+0/3: ->" runat="server" ID="ThemeMaster" />" [1]
[CompletionsProposalSource Information] Request completions: 						<|+0/15: No completions found
[CompletionsProposalSource Information] Request completions: 						<|+0/15: ->">" [1]
[CompletionsProposalSource Information] Request completions: 					    <|+0/44: No completions found
[CompletionsProposalSource Information] Request completions: 					    <|+0/44: ->">" [1]
[CompletionsProposalSource Information] Request completions: 					    >|+0/50: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr>|+0/53: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr><|+0/54: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr><|+0/54: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr></|+0/55: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr></|+0/55: No completions found
[CompletionsProposalSource Information] Request completions: 					    <tr></tr>|+0/56: No completions found
[CompletionsProposalSource Information] Request completions: |+28/58: ->"<td>" [1]
[CompletionsProposalSource Information] Request completions: |+32/61: ->"<strong>Edit Theme.master:</strong>" [1]
[CompletionsProposalSource Information] Request completions: |+32/64: ->"</td>" [1]
[CopilotClientManager Information] stderr: (node:38252) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 change listeners added to [EventEmitter]. Use emitter.setMaxListeners() to increase limit
[CopilotClientManager Information] stderr: (Use `copilot-language-server --trace-warnings ...` to show where the warning was created)
[CompletionsProposalSource Information] Request completions: |+28/67: ->"<td>" [1]
```

**Reply by mfsbo** [reply]
Installed 17.12.2 ([release notes](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#17.12.2))
Started and opend an aspx file to edit. Copilot log showed this
```
ogging to: C:\user\username\AppData\Local\Temp\VSGitHubCopilotLogs\20241127_173053.720_VSGitHubCopilot.chat.log
[Conversations Information] Copilot chat version 17.12.38-rc+719e1607cc.RR (17.12.38.29086). VS: VisualStudio.17.Release/17.12.2+35521.163. Session: d9b43ffb-b160-4e81-b577-4134d90d829b. 
[Conversations Error] Loading GitHub account information...
[Conversations Information] Need to get an auth token.
[Conversations Information] Status Code: 200, Reason Phrase: OK
[Conversations Information] Response Content: HasToken: True, ChatEnabled: True, ExpiresAt: 1732730455, RefreshIn: 1500, SKU: yearly_subscriber, CopilotExclusion: False
[Conversations Information] Copilot badge status set to: OK
[Conversations Information] [Conversations] Out-of-process service (PID: 11592) finished initializing.
[Conversations Information] [ChunkIndex] Index db path: C:\Root\Projects\github\repo\.vs\something-System\CopilotIndices\17.12.38.29086\CodeChunks.db
[CopilotClientManager Information] LSP: Debug Server: http://localhost:8180
[CopilotClientManager Information] LSP: Process 30504
[CopilotClientManager Information] LSP: Activated (16099925)
[CopilotClientManager Information] Initial auth status: OK
[CompletionsProposalSource Information] Request completions: |+24/2: No completions found
[CompletionsProposalSource Information] Request completions: 						<|+0/3: No completions found
[CompletionsProposalSource Information] Request completions: 						<|+0/3: No completions found
[CompletionsProposalSource Information] Request completions: 						<t|+0/4: No completions found
[CompletionsProposalSource Information] Request completions: 						<tr|+0/5: ->">" [1]
```
Putting `<t` produced some complettion but it produces a chunk of code shown below and did not close the td. I try to close the tag and it starts typing double.
![image](https://github.com/user-attachments/assets/673ded5f-0944-449a-a300-ac1d84a7c312)
From then onward it kept producing this 
![image](https://github.com/user-attachments/assets/56898ebc-7707-4610-83f8-88d9bc8d3db2)

**Comment by sitefinitysteve** [comment]
Top right->Github Copilot->Settings->Enable Copilot Completions

Turn that off and it's fixed... but COME ON VS TEAM, push out a fix please, this is unacceptable for what we pay for the IDE

**Reply by Edddy** [reply]
For the IDE and for Copilot
